pkgdb/src/lock-flake-installable.cc

```diff
@@ -550,6 +552,43 @@ from_json( const nlohmann::json & jfrom, LockedInstallable & to )
 }
 
 
+void
+prim_lockFlakeInstallable( nix::EvalState &  state,
+                           const nix::PosIdx pos,
+                           nix::Value **     args,
+                           nix::Value &      value )
+{
+
+
+  nix::NixStringContext context;
+
+  if ( args[0]->isThunk() && args[0]->isTrivial() )
+    {
+      state.forceValue( *args[1], pos );
+    }
+
+  auto state2 = nix::make_ref<nix::EvalState>( nix::SearchPath(),
+                                               state.store,
+                                               state.store );
+
+  auto system = nix::settings.thisSystem.get();
+
+  auto lockedInstallable
+    = lockFlakeInstallable( state2, system, std::string( args[0]->str() ) );
+
+  auto lockedInstallableJson = nlohmann::json( lockedInstallable );
+  nix::parseJSON( state, lockedInstallableJson.dump(), value );
+}
+
+
+static const nix::RegisterPrimOp
```

Comment:
Doesn't really matter, but could we use registerCommand to create a subcommand intead of creating a primop?

Reply:
Yeah I was thinking it might be slightly clearer but doesn't matter much especially if it's more work